### ISSUE TITLE
Site not loading after update to 2.0.4

### ISSUE THREAD

**Issue by bentrax** [modxbot/migrate#2502]
bentrax created Redmine issue ID 2502

I had a 2.0.3 installation with some content and a second resource group called "restricted".

After the update the manager loaded with no problems but the web did not.

I got an error pointing to /core/model/modx/modaccessibleobject.class.php line 26 saying that getPrimaryKey() is not possible without an object or something like that.
Sorry don't have the real error message anymore since I deleted the resource group and recreated it and set all the pages that have been in the previous group to the new one.
After that my site is working again.

Didn't report at once since my primary goal was to get my production site up again.

**Comment by bentrax** [comment]
bentrax submitted:

I tried to get further with my acces policies on the site and there the error comes back.

Fatal error: Call to a member function getPrimaryKey() on a non-object in /www/htdocs/w00c9310/www/core/model/modx/modaccessibleobject.class.php on line 26

As previously mentioned I have two resource groups "public" (the default one) and "restricted" and one resource that is in the "restricted" group.
I have a test user other than the administrator set that he is able to view resources in the "restricted" group by setting him to that resource group, minimal neede role is "Member - 9999", access policy is "Load only" and context is "web".

After I assign that to the user the site brings the error.

When I remove that from the user the error is gone, I can see my site but my menu shows up the entry for the resource that is only assinged to "restricted".

In 2.0.1 this was working as intended, only after login the menu was there and I was able to access the resource, with no login no access.

I know, I think I have to assing the "public" resource group to the (anonymous) user group to make them not see all what is for member in web but when I try to do this same error occurs.

Maybe the title of the bug should be changed to "Assigning resource groups to users make site not loading" and the priority for me is now blocker.
